login/profile.go

```diff
@@ -0,0 +1,129 @@
+	FirstName                  *string                        `json:"firstName"`
+	LastName                   *string                        `json:"lastName"`
+	Email                      *string                        `json:"email"`
+	Attributes                 *KeycloakUserProfileAttributes `json:"attributes"`
+	DisableableCredentialTypes []*string                      `json:"disableableCredentialTypes"`
+	RequiredActions            []interface{}                  `json:"requiredActions"`
+}
+
+// NewKeycloakUserProfile creates a new keycloakUserProfile instance.
+func NewKeycloakUserProfile(firstName *string, lastName *string, email *string, attributes *KeycloakUserProfileAttributes) *KeycloakUserProfile {
+	return &KeycloakUserProfile{
+		FirstName:  firstName,
+		LastName:   lastName,
+		Email:      email,
+		Attributes: attributes,
+	}
+}
+
+// UserProfileService describes what the services need to be capable of doing.
+type UserProfileService interface {
+	Update(keycloakUserProfile *KeycloakUserProfile, accessToken string, keycloakProfileURL string) error
+	Get(accessToken string, keycloakProfileURL string) (*KeycloakUserProfileResponse, error)
+}
+
+// KeycloakUserProfileClient describes the interface between platform and Keycloak User profile service.
+type KeycloakUserProfileClient struct {
+	client *http.Client
+}
+
+// NewKeycloakUserProfileClient creates a new KeycloakUserProfileClient
+func NewKeycloakUserProfileClient() *KeycloakUserProfileClient {
+	return &KeycloakUserProfileClient{
+		client: http.DefaultClient,
+	}
+}
+
+// Update updates the user profile information in Keycloak
+func (userProfileClient *KeycloakUserProfileClient) Update(keycloakUserProfile *KeycloakUserProfile, accessToken string, keycloakProfileURL string) error {
+	body, err := json.Marshal(keycloakUserProfile)
+	if err != nil {
+		return errors.NewInternalError(err.Error())
+	}
+
+	req, err := http.NewRequest("POST", keycloakProfileURL, bytes.NewReader(body))
+	if err != nil {
+		return errors.NewInternalError(err.Error())
+	}
+	req.Header.Add("Authorization", "Bearer "+accessToken)
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := userProfileClient.client.Do(req)
+	if err != nil {
+		return errors.NewInternalError(err.Error())
+	}
+	defer resp.Body.Close()
+	return nil
+}
+
+//Get gets the user profile information from Keycloak
+func (userProfileClient *KeycloakUserProfileClient) Get(accessToken string, keycloakProfileURL string) (*KeycloakUserProfileResponse, error) {
+
+	keycloakUserProfileResponse := KeycloakUserProfileResponse{}
+
+	req, err := http.NewRequest("GET", keycloakProfileURL, nil)
+	if err != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	req.Header.Add("Authorization", "Bearer "+accessToken)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "application/json, text/plain, */*")
+
+	resp, err := userProfileClient.client.Do(req)
+	defer resp.Body.Close()
```

Comment:
> // The request Body, if non-nil, will be closed by the underlying
> // Transport, even on errors.

So, just don't close the Body. We don't have to. And if err!=nil just return an error and do not proceed.